demisto_sdk/commands/download/downloader.py

```diff
@@ -1,741 +1,978 @@
                 return 1
 
-        return 0
+            output_path = Path(self.output_pack_path)
 
-    def verify_flags(self) -> bool:
-        """
-        Verifies that the flags configuration given by the user is correct
-        :return: The verification result
-        """
-        is_valid = True
-        if not self.list_files:
-            output_flag, input_flag = True, True
-            if not self.output_pack_path:
-                output_flag = False
-                logger.info("[red]Error: Missing option '-o' / '--output'.[/red]")
-            if not self.input_files:
-                if not self.all_custom_content and not self.regex:
-                    input_flag = False
-                    logger.info("[red]Error: Missing option '-i' / '--input'.[/red]")
-            if not input_flag or not output_flag:
-                is_valid = False
-
-        if self.download_system_item and not self.system_item_type:
-            logger.info(
-                "[red]Error: Missing option '-it' / '--item-type', "
-                "you should specify the system item type to download.[/red]"
+            if not self.verify_output_path(output_path=output_path):
+                return 1
+
+            if self.init:
+                output_path = self.initialize_output_path(root_folder=output_path)
+
+            if self.download_system_items:
+                if not self.input_files:
+                    logger.error(
+                        "Error: Missing required parameter for downloading system items: '-i' / '--input'."
+                    )
+                    return 1
+
+                if not self.system_item_type:
+                    logger.error(
+                        "Error: Missing required parameter for downloading system items: '-it' / '--item-type'."
+                    )
+                    return 1
+
+                content_item_type = self.system_item_type
+                downloaded_content_objects = self.fetch_system_content(
+                    content_item_type=content_item_type
+                )
+
+            else:  # Custom content
+                if not any(
+                    (self.input_files, self.download_all_custom_content, self.regex)
+                ):
+                    logger.error(
+                        "Error: No input parameter has been provided "
+                        "('-i' / '--input', '-r' / '--regex', '-a' / '--all)."
+                    )
+                    return 1
+
+                all_custom_content_data = self.download_custom_content()
+                all_custom_content_objects = self.parse_custom_content_data(
+                    custom_content_data=all_custom_content_data
+                )
+
+                # Filter custom content so that we'll process only downloaded content
+                downloaded_content_objects = self.filter_custom_content(
+                    custom_content_objects=all_custom_content_objects
+                )
```

Comment:
```suggestion
                downloaded_content_items= self.filter_custom_content(
                    custom_content_objects=all_custom_content_objects
                )
```
doesn't make more sense?

Reply:
Same explanation as https://github.com/demisto/demisto-sdk/pull/3295/files#discussion_r1365556329